test/search/real_search.jl

```diff
@@ -0,0 +1,91 @@
+using JSON
+
+# Load the real search index from test examples (already built!)
+function load_real_search_index()
+    # Use the example search index that's already built and tested
+    search_index_path = joinpath(@__DIR__, "../examples/builds/html/search_index.js")
+    
+    if !isfile(search_index_path)
+        error("Search index not found at: $search_index_path")
+    end
+    
+    # Read and parse the JavaScript file
+    content = read(search_index_path, String)
+    
+    # Find the JSON data after "var documenterSearchIndex = "
+    json_start = findfirst("var documenterSearchIndex = ", content)
+    if json_start === nothing
+        error("Invalid search index format: missing variable declaration")
+    end
+    
+    # Extract JSON content (everything after the variable declaration)
+    json_content = content[last(json_start) + 1:end]
+    
+    # Parse the JSON
+    parsed = JSON.parse(json_content)
+    return parsed["docs"]  # Return just the docs array
+end
+
+# Simple function that uses the existing search.js with real search data
+function real_search(query::String)
+    # Load the real search index automatically
+    search_index_data = load_real_search_index()
+    
+    # Create a Node.js script that recreates the search logic from search.js
+    wrapper_js = """
```

Comment:
This one is pretty big. Might be better to have this as a `.js` file that we `read(..., String)` in.